resources/indexer.py

```diff
@@ -18,290 +19,287 @@
-    # Ex: Level2_IC86.2017_data_Run00130567_Subrun00000000_00000280.i3.zst
-    # Ex: Run00125791_GapsTxt.tar
-    r = file.name.split('Run')[1]
-    run = int(r.split('_')[0])
-    return run
-
-
-def _disect_filename(file):
-    """Return year/run/subrun/part number from filename"""
-
-    def _get_year_from_year(file):
-        # Ex: Level2_IC86.2017_[...]
-        y = file.name.split('.')[1]
-        year = int(y.split('_')[0])
-        return year
-
-    run = _get_run_number(file)
+    @staticmethod
+    def name_to_year(name):
+        """Return the year of the season start for the season's name."""
+        if name:
+            for season_year, season_name in IceCubeSeason.SEASONS.items():
+                if season_name == name:
+                    return int(season_year)
+            raise Exception(f"No season found for {name}.")
+        else:
+            return None
+
+    @staticmethod
+    def year_to_name(season_year):
+        """Return the season's name for the year of the season start."""
+        if season_year:
+            try:
+                return IceCubeSeason.SEASONS[str(season_year)]
+            except KeyError:
+                raise Exception(f"No season found for {season_year}.")
+        else:
+            return None
+
+
+class ProcessingLevel:
+    """Wrapper static class encapsulating processing-level parsing logic and levels constants."""
+    PFRaw = "PFRaw"
+    PFFilt = "PFFilt"
+    PFDsT = "PFDsT"
+    L2 = "L2"
+    L3 = "L3"
+    L4 = "L4"
+
+    @staticmethod
+    def from_path(dir_path):
+        """Return the processing level parsed from the dir path, case insensitive."""
+        # Ex: /data/exp/IceCube/2013/filtered/level2/1229/Run00123579_0/
+        search_strings = {
+            "PFRaw": ProcessingLevel.PFRaw,
+            "PFFilt": ProcessingLevel.PFFilt,
+            "PFDsT": ProcessingLevel.PFDsT,
+            "Level2": ProcessingLevel.L2,
+            "Level3": ProcessingLevel.L3,
+            "Level4": ProcessingLevel.L4
+        }
+        dir_path_upper = dir_path.upper()
+        for string, level in search_strings.items():
+            if string.upper() in dir_path_upper:
+                return level
+        raise Exception(f"Cannot detect processing level, {dir_path}.")
+
+
+class BasicFileMetadata:
+    """Metadata for basic files"""
+    file = None
+    site = ""
+
+    def __init__(self, file, site):
+        self.file = file
+        self.site = site
+
+    def generate(self):
+        """Gather the file's metadata."""
+        metadata = dict()
+        metadata['logical_name'] = self.file.path
+        metadata['checksum'] = {'sha512': self.sha512sum()}
+        metadata['file_size'] = self.file.stat().st_size
+        metadata['locations'] = [{'site': self.site, 'path': self.file.path}]
+        metadata['create_date'] = date.fromtimestamp(os.path.getctime(self.file.path)).isoformat()
+        return metadata
```

Comment:
I would rewrite this in a more literal style:
```
def generate(self):
    """Gather the file's metadata."""
    return {
        'logical_name': self.file.path,
        'checksum': {
            'sha512': self.sha512sum(),
        },
        'file_size': self.file.stat().st_size,
        'locations': [
            {
                'site': self.site,
                'path': self.file.path,
            },
        ],
        'create_date': date.fromtimestamp(os.path.getctime(self.file.path)).isoformat(),
    }
```

But I may also be scarred from years of JavaScript/CoffeeScript, so feel free to ignore.

Reply:
I think the question comes down to if you ever want to optionally include a field, or if these fields are always present.  I'm fine with either style, though I will note that the proper way to create an empty dict is: `metadata = {}`

Reply:
I like the compact style since there isn't a whole lot going on. I'll split the difference and replace `dict()` with `{}`.